packages/block-editor/src/components/block-navigation/use-block-navigation-client-ids.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { isClientIdSelected } from './utils';
+import { store as blockEditorStore } from '../../store';
+
+export default function useBlockNavigationClientIds(
+	blocks,
+	showOnlyCurrentHierarchy,
+	selectedBlockClientIds,
+	__experimentalPersistentListViewFeatures
+) {
+	return useSelect(
+		( select ) => {
+			const {
+				getBlockHierarchyRootClientId,
+				getSelectedBlockClientId,
+				getSelectedBlockClientIds,
+				__unstableGetClientIdsTree,
+				__unstableGetClientIdWithClientIdsTree,
+			} = select( blockEditorStore );
+
+			let selectedClientIds;
+			if ( selectedBlockClientIds ) {
+				selectedClientIds = selectedBlockClientIds;
+			} else if ( __experimentalPersistentListViewFeatures ) {
+				selectedClientIds = getSelectedBlockClientIds();
+			} else {
+				selectedClientIds = getSelectedBlockClientId();
+			}
+
+			let clientIdsTree = blocks;
```

Comment:
I think we can remove the assignment here since we are doing it again right below.
```suggestion
			let clientIdsTree;
```

Reply:
Whoopsie, that was a leftover from a WIP state!